smpperf.go

```diff
@@ -175,18 +191,31 @@ func (s *SMPPerf) SendMessages() {
 			}
 			time.Sleep(r.Delay())
 			currTransceiver = (currTransceiver + 1) % s.NumSessions
-			go func(transceiverIndex int) {
-				sm, err := transceivers[transceiverIndex].Submit(req)
-				if err != nil {
-					go sendErrorCount.Increment()
-				} else {
-					transceiverID := strconv.Itoa(transceiverIndex)
-					msgIDToTransceiverID.Set(sm.RespID(), transceiverID)
-					submittedCount.Increment()
-				}
-			}(currTransceiver)
+			if transceivers[currTransceiver].err == nil {
+				// here we send the message
+				go func(transceiverIndex int) {
+					sm, err := transceivers[transceiverIndex].transceiver.Submit(req)
+					if err != nil {
+						if err == smpp.ErrNotConnected {
+							i--
+							go connErrorCount.Increment()
+						} else {
+							go sendErrorCount.Increment()
+						}
+					} else {
+						transceiverID := strconv.Itoa(transceiverIndex)
+						msgIDToTransceiverID.Set(sm.RespID(), transceiverID)
+						submittedCount.Increment()
+					}
+				}(currTransceiver)
+				i++
```

Comment:
Why counting the message as successful here and not below `submittedCount.Increment()`?